sycl/include/sycl/detail/cg.hpp

```diff
@@ -140,6 +140,8 @@ class CG {
   }
   std::vector<detail::EventImplPtr> &getEvents() { return MData.MEvents; }
 
+  virtual void printDot(std::ostream &Stream) const {}
+
```

Comment:
This is an ABI change which fails the `abi/vtable.cpp` test, which i'm not sure is an [ABI breaking](https://github.com/intel/llvm/blob/sycl/sycl/doc/developer/ABIPolicyGuide.md#intro) or not. If it is breaking, we'll have a problem getting this merged into DPC++ as the ABI breaking windows has just closed and not sure when the next one is.

Reply:
`check-sycl` or `./bin/llvm-lit --param sycl_devices="ext_oneapi_level_zero:gpu" -sv ~/Development/dpcpp/sycl/test/abi`

Reply:
Adding new virtual function to a non-leaf class is actually an ABI breaking change.

Reply:
I changed the implementation to avoid ABI breaking.